components/transferForm.tsx

```diff
@@ -506,12 +510,33 @@ function TransferForm(
         setFormState(values);
         setLoading(true);
         try {
-          let cc = await state.connection.wallet.at(props.address);
+          const cc = await state.connection.wallet.at(props.address);
+          const versioned = VersionedApi(props.contract.version, props.address);
+          const proposals: proposals = { transfers: values.transfers };
 
-          let versioned = VersionedApi(props.contract.version, props.address);
           setTimeoutAndHash(
-            await versioned.submitTxProposals(cc, state.connection, values)
+            await versioned.submitTxProposals(cc, state.connection, proposals)
           );
+
+          const storage = toStorage(
+            versioned.version,
+            await cc.storage(),
+            BigNumber(0)
+          );
+          const proposal_id = storage.proposal_counter;
+
+          if (values.signImmediatelyFlag) {
+            setTimeoutAndHash(
+              await versioned.signProposal(
+                cc,
+                state.connection,
+                proposal_id,
```

Comment:
Same than before on styling and reusing the same code

Reply:
Done